Pagination, switched to apollo hooks.

snacksncode opened the pull request:
**Changes:** 
- Added pagination
- Switched from `<Query>` to `useQuery()`
- Added markdown to blogEntry's subtitle
- Minor changes

Darex1991 commented on components/Footer/Footer.jsx:
```diff
@@ -17,111 +21,106 @@ const articlesQuery = gql`
+            width="340"
+            title="facebook"
+            height="120"
+            style={{ border: 'none', overflow: 'hidden' }}
```
please move this to global styles

Darex1991 commented on components/NavigationButtons/navigationButtonsStyles.scss:
```diff
@@ -0,0 +1,10 @@
+@import '../../styles/variables';
+
+.pagination-buttons {
+  &__container {
```
please merge it with parent

Darex1991 commented on components/SideBar/sideBarStyles.scss:
```diff
@@ -7,7 +7,7 @@
 }
 
-@media (min-width: $max-width-cap) {
+@media (min-width: 992px) {
```
please use variables

Darex1991 commented:
@snacksncode navigation is wrong. Again you didn't check mockups!
<img width="1440" alt="Screenshot 2020-06-29 at 10 25 58" src="https://user-images.githubusercontent.com/5881611/85990941-05940680-b9f3-11ea-8097-3653d5094586.png">

snacksncode commented:
Oh there is navigation there. Didn't look at it. Fixable

snacksncode commented:
- added custom 404 page
- changed name from `NavigationButtons` to `PaginationButtons` to more clearly represent component
- pagination buttons styled according to mock-up
- minor changes in error handling

Darex1991 commented:
1) <img width="997" alt="Screenshot 2020-06-29 at 23 08 45" src="https://user-images.githubusercontent.com/5881611/86056331-96073100-ba5d-11ea-84d9-eb2db006408d.png">
2) focused btn shoudn't be blue
<img width="465" alt="Screenshot 2020-06-29 at 23 07 56" src="https://user-images.githubusercontent.com/5881611/86056354-9b647b80-ba5d-11ea-94bb-36113c39243c.png">
